### PR TITLE
feat: more human friendly makefile output

### PR DESCRIPTION
**Pull request by leslie-tsang**
* better makefile output
* more flexible docker ENV inject support
https://github.com/NGX-AI/apisix-docker/blob/0576345238eabb7ef4e07dbffe100722ba365f1d/Makefile#L35-L38

> before
```bash
Makefile rules:

    build-on-centos:      Build apache/apisix:xx-centos image
    build-on-alpine:      Build apache/apisix:xx-alpine image
    build-on-alpine-local:      Build apache/apisix:xx-alpine-local image
    push-on-centos:       Push apache/apisix:xx-centos image
    push-on-alpine:       Push apache/apisix:xx-alpine image
    build-on-alpine-cn:          Build apache/apisix:xx-alpine image (for chinese)
    build-all-in-one:            Build All in one Docker container for Apache APISIX
    save-centos-tar:      tar apache/apisix:xx-centos image
    save-alpine-tar:      tar apache/apisix:xx-alpine image
    build-dashboard:    Build apache/dashboard:tag image
    push-dashboard:     Push apache/dashboard:tag image
    save-dashboard-tar:      tar apache/apisix-dashboard:tag image
    help:                 Show Makefile rules
```
> after
```bash
[ info ] Makefile rules:

    make build-on-centos           : Build apache/apisix:xx-centos image
    make build-on-alpine           : Build apache/apisix:xx-alpine image
    make build-on-alpine-local     : Build apache/apisix:xx-alpine-local image
    make push-on-centos            : Push apache/apisix:xx-centos image
    make push-on-alpine            : Push apache/apisix:xx-alpine image
    make build-on-alpine-cn        : Build apache/apisix:xx-alpine image (for chinese)
    make build-all-in-one          : Build All in one Docker container for Apache APISIX
    make save-centos-tar           : tar apache/apisix:xx-centos image
    make save-alpine-tar           : tar apache/apisix:xx-alpine image
    make build-dashboard           : Build apache/dashboard:tag image
    make push-dashboard            : Push apache/dashboard:tag image
    make save-dashboard-tar        : tar apache/apisix-dashboard: tag image
    make help                      : Show Makefile rules
```
Signed-off-by: Leslie <leslie.tsang@icloud.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -24,69 +28,131 @@ APISIX_DASHBOARD_VERSION ?= 2.8
...
+
+
+define func_echo_status
+	printf "[$(_color_blue) info $(_color_wipe)] %s\n" $(1)
+endef
+
+
+define func_echo_warn_status
+	printf "[$(_color_yellow) info $(_color_wipe)] %s\n" $(1)
+endef
```
Comment (imjoey) [inline Makefile:54]: Hi, these two functions seem not used? Any reason for this?

Comment (leslie-tsang) [reply Makefile:54]: These two functions are reserved for better extend support

Comment (leslie-tsang) [reply Makefile:54]: > Hi, these two functions seem not used? Any reason for this?

eg: more clear run out
```bash
### build-on-centos:      Build apache/apisix:xx-centos image
build-on-centos:
	@$(call func_echo_status, "$@ -> [ Start ]")
	docker build -t $(IMAGE_NAME):$(APISIX_VERSION)-centos -f ./centos/Dockerfile .
	@$(call func_echo_success_status, "$@ -> [ Done ]")
```
> output
```bash
[ info ] build-on-centos -> [ Start  ]
docker build output ...
[ info ] build-on-centos -> [ Done  ]
```

Comment (imjoey) [reply Makefile:54]: @leslie-tsang Got you, thanks.

**Review by imjoey — APPROVED** [review]
/lgtm

**Review by tokers — APPROVED** [review]

**Review by nic-chen — APPROVED** [review]

**Pull request merged**